### ISSUE TITLE
[Design System] Offer an `aria` property across components for further configuration options

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-components#13912]
**monday.com sync:** #11243164856

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Which Component

N/A

### Description

We should evaluate the use of a new `aria` property across components, similar to the implementation to the `action` component with #10706 for `5.0` offering options that should be supported for each use case.

For instance, in `action`'s implementation [the following are offered](https://developers.arcgis.com/calcite-design-system/components/action/#api-reference-properties-aria):

https://github.com/Esri/calcite-design-system/blob/3caa202b4ef6306ec3b68a3ae82dd9506d261d72/packages/components/src/components/action/action.tsx#L72-L85

Calcite should evaluate using selection-type components across the design system for consistency and configuration options that work for more scenarios.

- [x] `action` - #10706
- [ ] `button` - #13655
- [ ] `card`
- [ ] `carousel-item`
- [ ] `chip`
- [ ] `dropdown`
- [ ] `dropdown-item`
- [ ] `flow-item`
- [ ] `list-item`
- [ ] `split-button`
- [ ] `stepper-item`
- [ ] `swatch`
- [ ] `tab`
- [ ] `tab-title`
- [ ] `tile`
- [ ] `tree-item`

cc @isaacbraun for additional 👀 if there could be additional considerations for the above

### Acceptance Criteria

- [ ] Consistent use of aria-named attributes across components
- [ ] Ensure configuration of aria-named attributes are available for components

### Relevant Info

For alignment across the design system with #10706, and with a team request for `button` alignment with #13655

### Example Use Case

```js
let actionElExpanded = false;

actionEl.addEventListener("click", () => {
  actionElExpanded = !actionElExpanded;
  actionEl.aria = {
    expanded: actionElExpanded
  }
});
```

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (dev)
